include/openenclave/edger8r/common.h

```diff
@@ -99,7 +99,7 @@ OE_INLINE oe_result_t oe_add_size(size_t* total, size_t size)
  * buffer. Make sure that the buffer has enough space.
  */
 #define OE_SET_OUT_POINTER(argname, argsize, argtype)                        \
-    if (pargs_in->argname)                                                   \
+    do                                                                       \
```

Comment:
What is the need for removing this check?

Previously, `pargs_in->argname` is set to a non-null value only if the original value was a non-null value.
With this change, `pargs_in->argname` is always set to a non-null value.

Reply:
This is because only first-level arguments are non-null. This check would block the initialization for the nested arguments. Therefore, I move the check logic to https://github.com/openenclave/openenclave/pull/2427/files#diff-e7bae2bcdd6045dc447a7f504b89b903R18.

The current behavior works as follow:
For the first-level arg, the generated check will be:
`if (pargs_in->s) OE_SET_OUT_POINTER(s, ...);`
For the nested args, the checks become:
`if (pargs_in->s && !pargs_in->s->ptr) OE_SET_OUT_POINTER(s->ptr, ...);`
`if (pargs_in->s && pargs_in->s->ptr && !pargs_in->s->ptr->tail) OE_SET_OUT_POINTER(s->ptr->tail, ...);`